### PR TITLE
feat(riak): add riak chart

### PR DESCRIPTION
**Pull request by bacongobbler**

### CODE REVIEW

**Comment by bacongobbler** [comment]
Note that users still need to run a few commands as per https://github.com/deis/riak#usage, but if HA isn't a concern then I can strip away everything but the pod manifest if that makes sense.

**Comment by gabrtv** [comment]
@bacongobbler let's change the path from `riak` to `riak-cluster` for convention.

**Comment by bacongobbler** [comment]
done :)

File: riak-cluster/manifests/riak-discovery-service.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - port: 8098
+      targetPort: 8098
+      name: protobuf
+      protocol: TCP
+    - port: 8099
+      targetPort: 8099
+      name: handoff
+      protocol: TCP
+  selector:
+    name: riak
```
Comment (gabrtv) [inline riak-cluster/manifests/riak-discovery-service.yaml]: Let's change this to `provider=riak` per the labels guidance in https://github.com/deis/helm/blob/master/docs/labels.md#provider-label

File: riak-cluster/manifests/riak-rc.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  labels:
+    heritage: deis
+spec:
+  replicas: 5
+  selector:
+    name: riak
+  template:
+    metadata:
+      labels:
+        name: riak
```
Comment (gabrtv) [inline riak-cluster/manifests/riak-rc.yaml]: `provider: riak`

**Comment by bacongobbler** [comment]
hold off on merging this one; I'm going to provide both worlds:
- HA Riak (current work)
- standalone Riak pod (same as other examples)

File: riak-cluster/manifests/riak-service.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    - port: 8087
+      targetPort: 8087
+      name: http
+      protocol: TCP
+    - port: 8098
+      targetPort: 8098
+      name: protobuf
+      protocol: TCP
+  selector:
+    name: riak
```
Comment (gabrtv) [inline riak-cluster/manifests/riak-service.yaml]: `provider: riak`

**Comment by technosophos** [comment]
Dude, you are seriously ripping it up with these! Good example of using the `provider` label.

**Comment by bacongobbler** [comment]
I dropped the clustered riak chart for now while trying to debug what's wrong with it so we can get this one out the door.

**Comment by mboersma** [comment]
Tested with helm 0.1.0, everything looks good and I can curl the server 8098 endpoint inside the cluster. `kubectl logs riak` doesn't show anything, but maybe it wouldn't? Looks good to me.

**Comment by bacongobbler** [comment]
Yeah that's something I need to fix up at some point. By default riak logs to a file inside the container. deis/riak was inspired by https://github.com/hectcastro/docker-riak which logs to /var/log/riak. I just need to tweak a few things inside https://github.com/deis/riak/blob/master/rootfs/bin/boot to start riak with logs in the foreground, though most of my attempts to do so have failed since I have to background the process.

One way is to use the same technique we use in deis/router but I'd rather do something more native instead of a goroutine piping logs to stdout/stderr.

**Pull request merged**